### PR TITLE
fix(ToolbarMethods) Selecting list/quote option while link popup is open crashes demo - #266

### PR DESCRIPTION
**Pull request by 98lenvi**
# Issue #266 
Now the web app doesn't crash when list options are selected when the link popover is open

### Changes
- In `ToolbarMethods.js`
  - In `transformParagraphToList` method, I have added one more condition check, ie of the text selection is only a link, The behavior of the popover is that it only opens when the selection is only a link. 
  - If the selection is only a link, then the usual conversion of the paragraph to list doesn't take place

### Flags
- Can be improved If I figure out a way to increase the selection area when the selection is only a list

### Related Issues
- Issue #266

### CODE REVIEW

**Comment by elit-altum** [comment]
@shrey-sachdeva2000 weren't you working on this ?

**Comment by 98lenvi** [comment]
> @shrey-sachdeva2000 weren't you working on this ?

It looked like he just raised this issue, so I thought to work on it. Even if he started, I'm sorry about that. I think we can work together to find a fix for this.

**Comment by shrey-sachdeva2000** [comment]
> > @shrey-sachdeva2000 weren't you working on this ?
> 
> It looked like he just raised this issue, so I thought to work on it. Even if he started, I'm sorry about that. I think we can work together to find a fix for this.

No worries, please go ahead if you've found a fix!

**Comment by irmerk** [comment]
You've carried over commits from the branch in the PR here: https://github.com/accordproject/markdown-editor/pull/267

**Comment by 98lenvi** [comment]
@irmerk, My bad. I've removed the unnecessary commits, Please have a look at it now!

**Comment by irmerk** [comment]
Nice! It looks like this solves it except for the block quote button.

**Comment by DianaLease** [comment]
> @irmerk - the block quote button seems to work on this when I try!

Nevermind, I misunderstood the issue. It does _not_ work for the block quote button

**Comment by 98lenvi** [comment]
@irmerk @DianaLease, My bad. I forgot about block quotes, Please check it now, I have added the fix for block quotes as well.

**Comment by irmerk** [comment]
@Michael-Grover do you mind also having a go at testing this?

@98lenvi could you clean up all the commits to be in the format we use in the [DEVELOPERS](https://github.com/accordproject/markdown-editor/blob/master/DEVELOPERS.md#-git-commit-guidelines) guide and also remove `[WIP]`

**Comment by 98lenvi** [comment]
@irmerk, really sorry about the commit messages, will fix it as soon as possible!

**Comment by 98lenvi** [comment]
@irmerk, I've made the changes, Please check it whenever you are free.

File: src/FormattingToolbar/toolbarMethods.js [hunk 1]
```diff
@@ -194,9 +194,11 @@ export const transformListToBlockQuote = (editor, type, value) => {
  */
 /* eslint no-unused-expressions: 0 */
 export const transformPtoBQSwap = (editor, type) => {
-  isSelectionInput(editor.value, CONST.BLOCK_QUOTE)
+  console.log('hi')
```
Comment (irmerk) [inline src/FormattingToolbar/toolbarMethods.js]: Get rid of this little guy.

**Comment by Michael-Grover** [comment]
> @Michael-Grover do you mind also having a go at testing this?

looks like it works to me

File: src/FormattingToolbar/toolbarMethods.js (hunk 1 above)
Comment (98lenvi) [reply src/FormattingToolbar/toolbarMethods.js]: oopsie

Comment (98lenvi) [reply src/FormattingToolbar/toolbarMethods.js]: The little guy is gone, Please check it now!

**Comment by 98lenvi** [comment]
@DianaLease, Please check it now, I've added the bugfix at a common place.

**Comment by irmerk** [comment]
This looks geat @98lenvi! This is a bit unusual, but we're implementing a `development` branch to keep `master` more stable and for actual releases. Could you open a new PR but open it to merge this branch to `development` instead of `master`? This looks done so we can just do some testing and get it merged.

I posted more information about this in slack and will be updating the `CONTRIBUTING` and `DEVELOPERS` guides soon.

**Comment by 98lenvi** [comment]
Please refer to #298

**Pull request closed without merging**